app/src/androidTest/java/com/example/atheneum/activities/FirebaseUIAuthActivitySignInTest.java

```diff
@@ -0,0 +1,238 @@
+                                                0)),
+                                0),
+                        isDisplayed()));
+        textView2.check(matches(withText("Sign in")));
+
+        ViewInteraction textView3 = onView(
+                allOf(withId(R.id.heading), withText("Welcome back!"),
+                        childAtPosition(
+                                childAtPosition(
+                                        IsInstanceOf.<View>instanceOf(android.widget.ScrollView.class),
+                                        0),
+                                0),
+                        isDisplayed()));
+        textView3.check(matches(withText("Welcome back!")));
+
+        ViewInteraction textInputEditText3 = onView(
+                allOf(withId(R.id.password),
+                        childAtPosition(
+                                childAtPosition(
+                                        withId(R.id.password_layout),
+                                        0),
+                                0)));
+        textInputEditText3.perform(scrollTo(), click());
+
+        ViewInteraction textInputEditText4 = onView(
+                allOf(withId(R.id.password),
+                        childAtPosition(
+                                childAtPosition(
+                                        withId(R.id.password_layout),
+                                        0),
+                                0)));
+        textInputEditText4.perform(scrollTo(), replaceText("logintester1"), closeSoftKeyboard());
+
+        ViewInteraction appCompatButton2 = onView(
+                allOf(withId(R.id.button_done), withText("Sign in"),
+                        childAtPosition(
+                                childAtPosition(
+                                        withClassName(is("android.widget.ScrollView")),
+                                        0),
+                                4)));
+        appCompatButton2.perform(scrollTo(), click());
+
+        // Added a sleep statement to match the app's execution delay.
+        // The recommended way to handle such scenarios is to use Espresso idling resources:
+        // https://google.github.io/android-testing-support-library/docs/espresso/idling-resource/index.html
+        try {
+            Thread.sleep(500);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+
+        ViewInteraction appCompatImageButton = onView(
+                allOf(withContentDescription("Open navigation drawer"),
+                        childAtPosition(
+                                allOf(withId(R.id.toolbar),
+                                        childAtPosition(
+                                                withClassName(is("android.support.design.widget.AppBarLayout")),
+                                                0)),
+                                1),
+                        isDisplayed()));
+        appCompatImageButton.perform(click());
```

Comment:
I'm getting it failing here with 
```
android.support.test.espresso.NoMatchingViewException: No views in hierarchy found matching: (with content description: is "Open navigation drawer" and Child at position 1 in parent (with id: 2131231032 and Child at position 0 in parent with class name: is "android.support.design.widget.AppBarLayout") and is displayed on the screen to the user)
```

Reply:
Can someone else run the test to check if this is just a one off or not?